src/cmap/auth/mongo_credentials.ts

```diff
@@ -146,15 +145,18 @@ export class MongoCredentials {
     }
 
     if (this.mechanism === AuthMechanism.MONGODB_OIDC) {
-      if (this.username && this.mechanismProperties.DEVICE_NAME) {
+      if (this.username && this.mechanismProperties.SERVICE_NAME) {
```

Comment:
since this was previously `DEVICE_NAME` which was a TS enum `'aws' | 'azure' | 'gcp'`, do we want to do an analogous thing for `SERVICE_NAME` or does the spec specifically say that it needs to allow an arbitrary string? If it does not require that we keep it arbitrary, perhaps it would be better to limit `SERVICE_NAME` to just `'aws'` (based on the validation below), and then add other options once they are supported

Reply:
Needs to be an arbitrary string for when Kerberos uses that field.